pkg/conversion/scheme.go

```diff
@@ -64,19 +60,21 @@ func NewScheme() *Scheme {
 		typeToGVK: map[reflect.Type][]unversioned.GroupVersionKind{},
 		converter: NewConverter(),
 		cloner:    NewCloner(),
-		// TODO remove this hard coded list.  As step one, hardcode it here so this pull doesn't become even bigger
-		InternalVersions: map[string]unversioned.GroupVersion{
-			"":                {},
-			"componentconfig": {Group: "componentconfig"},
-			"extensions":      {Group: "extensions"},
-			"metrics":         {Group: "metrics"},
-		},
-		MetaFactory: DefaultMetaFactory,
 	}
 	s.converter.nameFunc = s.nameFunc
 	return s
 }
 
+// AddInternalGroupVersion registers an internal version to the scheme
```

Comment:
Just a simple refactor of what was in master, I plan on moving it out in a
subsequent commit.  Glad you agree


On Dec 1, 2015, at 7:20 PM, Daniel Smith <notifications@github.com> wrote:

In pkg/conversion/scheme.go
<https://github.com/kubernetes/kubernetes/pull/17922#discussion_r46361277>:

>  	}
>  	s.converter.nameFunc = s.nameFunc
>  	return s
>  }
>
> +// AddInternalGroupVersion registers an internal version to the scheme

hm, shouldn't it be able to tell internal or not based on the
GroupVersion's version field? Why the separate method?

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/17922/files#r46361277>.